tools/lzcomp.c

```diff
@@ -220,7 +326,8 @@ unsigned short scan_backwards (const unsigned char * data, unsigned short limit,
   unsigned short position;
   for (position = 0; position < real_position; position ++) {
     if (data[position] != data[real_position]) continue;
-    for (current_length = 0; (current_length < limit) && (data[position - current_length] == data[real_position + current_length]); current_length ++);
+    for (current_length = 0; (current_length <= position) && (current_length < limit) &&
```

Comment:
For the record, this is the single-line fix. The rest are new features I added eventually (including improved compression), but I figured there was no reason to hold them back in pokecrystal.